pkg/controller/cronjob/cronjob_controller.go

```diff
@@ -104,19 +106,35 @@ func (jm *CronJobController) syncAll() {
 	// This guarantees that if we see any Job that got orphaned by the GC orphan finalizer,
 	// we must also see that the parent CronJob has non-nil DeletionTimestamp (see #42639).
 	// Note that this only works because we are NOT using any caches here.
-	jl, err := jm.kubeClient.BatchV1().Jobs(metav1.NamespaceAll).List(metav1.ListOptions{})
+	jobListFunc := func(opts metav1.ListOptions) (runtime.Object, error) {
+		return jm.kubeClient.BatchV1().Jobs(metav1.NamespaceAll).List(opts)
+	}
+	jlTmp, err := pager.New(pager.SimplePageFunc(jobListFunc)).List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
 		utilruntime.HandleError(fmt.Errorf("can't list Jobs: %v", err))
 		return
 	}
+	jl, ok := jlTmp.(*batchv1.JobList)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("expected type *batchv1.JobList, got type %T", jlTmp))
+		return
+	}
 	js := jl.Items
 	glog.V(4).Infof("Found %d jobs", len(js))
 
-	sjl, err := jm.kubeClient.BatchV1beta1().CronJobs(metav1.NamespaceAll).List(metav1.ListOptions{})
+	cronJobListFunc := func(opts metav1.ListOptions) (runtime.Object, error) {
+		return jm.kubeClient.BatchV1beta1().CronJobs(metav1.NamespaceAll).List(opts)
+	}
+	sjlTmp, err := pager.New(pager.SimplePageFunc(cronJobListFunc)).List(context.TODO(), metav1.ListOptions{})
```

Comment:
Rename from `sjl`/`sjlTmp` -> `cjl`/`cjlTmp` (the `s` refers to the former name for cronjobs, I think)?  It's actually probably more readable just to spell it out though - e.g. `jobs`, `cronjobs`, `cronjobList` etc

Reply:
Thanks! I was trying to minimize the change, but spelling out sounds good.